src/irr/asset/CPLYMeshWriter.cpp

```diff
@@ -53,12 +53,18 @@ static asset::E_FORMAT getCorrespondingIntegerFormat(asset::E_FORMAT _fmt)
 
 void flipNormalsAndPositions(asset::ICPUMeshBuffer* _mbuf, uint32_t ix)
 {
-    core::vectorSIMDf tmpValue;
+    auto flip = [&](const auto attributeIx, core::vectorSIMDf tmpValue = {})
+    {
 
-    _mbuf->getAttribute(tmpValue, 0, ix);
-    tmpValue[0] = -tmpValue[0];
-    _mbuf->getAttribute(tmpValue, 3, ix);
-    tmpValue[0] = -tmpValue[0];
+        if (_mbuf->getAttribute(tmpValue, attributeIx, ix))
+        {
+            tmpValue[0] = -tmpValue[0];
+            _mbuf->setAttribute(tmpValue, attributeIx, ix);
+        }
+    };
```

Comment:
in what world are you allowed to mutate the asset you're supposed to be writing to file!?

Reply:
to fix